### ISSUE TITLE
Stop using github.com/pkg/errors

### ISSUE THREAD

**Issue by dims** [kubernetes/kubernetes#103043]
We should be using Go native error wrapping (available since Go 1.13) (similar to what was done in https://github.com/containerd/console/pull/54)

NOTE : Please do not submit PRs for `cmd/kubeadm/app` though it is listed below.

code search : https://cs.k8s.io/?q=%22github.com%2Fpkg%2Ferrors%22&i=nope&files=&excludeFiles=vendor%2F&repos=kubernetes/kubernetes
ripgrep search : 
```
[dims@dims-a01 06:29] ~/go/src/k8s.io/kubernetes ⟩ rg '"github.com/pkg/errors"' | grep -v vendor/ | cut -f 1 -d ':' | cut -d '/' -f 1-3 | sort | uniq -c | sort
   1 cmd/kube-proxy/app
   1 cmd/kubeadm/test
   1 hack/conformance/check_conformance_test_requirements.go
   1 pkg/kubelet/pluginmanager
   1 pkg/kubelet/server
   1 pkg/util/ipvs
   1 pkg/util/netsh
   1 test/utils/deployment.go
   1 test/utils/pki_helpers.go
   2 pkg/kubelet/kubeletconfig
   2 staging/src/k8s.io
   2 test/e2e/windows
   3 cluster/images/conformance
   3 test/e2e/framework
   3 test/e2e/network
   4 test/e2e/storage
 120 cmd/kubeadm/app
```

If folks can pick up one line each and submit a PR, that would help get this turned around quickly. thanks!

BONUS: if someone can reuse one of the existing verify scripts under hack to ensure we don't regress, that would be awesome!

**Comment by dims** [comment]
@mcbenjemaa since you have contributed to k8s before and are in the github org, may be you can tackle the BONUS one? leave the easy stuff to new folks?

**Comment by mcbenjemaa** [comment]
@dims For sure!

**Comment by gdsoumya** [comment]
@dims would like to take this up

**Comment by neolit123** [comment]
Makes sense to cleanup sparse usage across the code base. But usage in kubeadm is intentional and %w is just not the same. We rather touch kubeadm for the time being.

**Comment by krak7602** [comment]
Pardon me for working on an assigned issue. I forgot to assign the issue . Also I have only finished one section. There are many remaining so the assigned person might not face issues.

**Comment by gdsoumya** [comment]
Taking up files :
```
   1 pkg/kubelet/server
   1 pkg/util/ipvs
   1 pkg/util/netsh
   1 pkg/kubelet/pluginmanager
```

**Comment by mcbenjemaa** [comment]
> If folks can pick up one line each and submit a PR, that would help get this turned around quickly. thanks!

@gdsoumya allow the others to submit their first PR.

**Comment by gdsoumya** [comment]
@mcbenjemaa each of these directories have a single file with 1/2 line changes mostly so I thought of taking up these in one go.

**Comment by Rahul-D78** [comment]
Hi I would like to pick
kubernetes/cluster/images/conformance/go-runner/main.go

**Comment by Rahul-D78** [comment]
Hi @dims 
`return fmt.Errorf(err, "failed to find absolute path for %v", resultsTarball)`
what about %v is it ok to replace with return `fmt.Errorof("failed to find absolute path for: %w", err, resultsTarball)`

**Comment by rajaSahil** [comment]
Hi, I would like to take up ` pkg/kubelet/pluginmanager`

**Comment by Haleygo** [comment]
Hi, I would like to take those under cmd without cmd/kubeadm/app :)
taking up :
cmd/kube-proxy/app
cmd/kubeadm/test

**Comment by rahulii** [comment]
Folks working on this should not mention `Fixes` in their PR , it will close this issue once that PR is merged, instead use `Part of `

**Comment by youhonglian** [comment]
Hi, I would like to take those under cmd without cmd/kubeadm/app :)
taking up :
pkg/kubelet/kubeletconfig
staging/src/k8s.io

**Comment by neolit123** [comment]
@Haleygo 

> cmd/kubeadm/test

we should not change kubeadm for the time being.

**Comment by Haleygo** [comment]
> @Haleygo
> 
> > cmd/kubeadm/test
> 
> we should not change kubeadm for the time being.

@neolit123 Sorry about that, I saw the conversation above says don't change files under cmd/kubeadm/app, but cmd/kubeadm/test is not under it, so I changed it.
I will change it back.

**Comment by ehashman** [comment]
> Taking up files :
> 
> ```
>    1 pkg/kubelet/server
>    1 pkg/util/ipvs
>    1 pkg/util/netsh
> ```

@gdsoumya Hi, please don't mix and match files between SIGs if possible. If you want to do node files, do

```
   1 pkg/kubelet/pluginmanager
   1 pkg/kubelet/server
   2 pkg/kubelet/kubeletconfig
```

If you want to do network, do

```
   1 cmd/kube-proxy/app
   1 pkg/util/ipvs
   1 pkg/util/netsh
   3 test/e2e/network
```

**Comment by gdsoumya** [comment]
@ehashman I have dropped the kubelet changes from my PR. Thanks for letting me know

**Comment by BenTheElder** [comment]
@dims are we _only_ using the wrapping feature? I know e.g. kind uses this package and will *not* be migrating because we also use the stacktrace feature.

**Comment by dims** [comment]
@BenTheElder yep that's fine. motivation was that if something is available in core golang, we should try to use it.

**Comment by BenTheElder** [comment]
er right, but have we validated that none of these packages depend on any of the functionality not provided by go?
changing error types is also a breaking change, some of those are in staging, but that's _probably_ fine.

**Comment by vivian-xu** [comment]
Hi, I would like to take up test/e2e/windows ~

**Comment by bhumijgupta** [comment]
@dims @ehashman I would like to take up the changes relating to sig/node


   ```
1 pkg/kubelet/pluginmanager
   1 pkg/kubelet/server
   2 pkg/kubelet/kubeletconfig
```

**Comment by ehashman** [comment]
> @dims @ehashman I would like to take up the changes relating to sig/node
> 
> ```
> 1 pkg/kubelet/pluginmanager
> 1 pkg/kubelet/server
> 2 pkg/kubelet/kubeletconfig
> ```

it's already done in https://github.com/kubernetes/kubernetes/pull/103059

**Comment by wzshiming** [comment]
Hi, I would like to take up test/e2e/network

**Comment by learner0810** [comment]
Hi, I would like to take up test/e2e/storage

**Comment by niulechuan** [comment]
I will take ``test/e2e/storage``

**Comment by muma378** [comment]
Taking up files ：）
   1 test/utils/deployment.go
   1 test/utils/pki_helpers.go

**Comment by wangyysde** [comment]
Hi, I would like to take up hack/conformance/check_conformance_test_requirements.go

**Comment by ash2k** [comment]
Would also be nice to drop `github.com/go-errors/errors`, which is a dependency of Kustomize but is used in a very few places only.

**Comment by ishangupta-ds** [comment]
Hi @dims @MadhavJivrajani 

Taking up files: 
1 test/conformance/image/go-runner/main.go
1 test/conformance/image/go-runner/tar.go
1 test/conformance/image/go-runner/tar_test.go

from https://cs.k8s.io/?q=%22github.com%2Fpkg%2Ferrors%22&i=nope&files=&excludeFiles=vendor%2F&repos=kubernetes/kubernetes

**Comment by ishangupta-ds** [comment]
@dims I am taking up the last file from - 

https://cs.k8s.io/?q=%22github.com%2Fpkg%2Ferrors%22&i=nope&files=&excludeFiles=vendor%2F&repos=kubernetes/kubernetes 

1 file - test/e2e/network/netopol/test_helper.go

all others are under `cmd/kubeadm/app`

PR #105250
